lib/Models/ArcGisFeatureServerCatalogItem.js

```diff
@@ -146,13 +212,68 @@ ArcGisFeatureServerCatalogItem.prototype.showOnSeparateMap = function(
   }
 };
 
+function setupUniqueValRenderer(renderer) {
+  const out = {};
+  for (var i = 0; i < renderer.uniqueValueInfos.length; i++) {
+    const val = renderer.uniqueValueInfos[i].value;
+    out[val] = renderer.uniqueValueInfos[i];
+  }
+  return out;
+}
+
+function convertEsriRendererToStyle(symbol) {
+  const styleObj = {};
+
+  const fill = symbol.color;
+
+  // Most things will have a fill but possibly not points that have an icon
+  if (defined(fill)) {
+    styleObj.fill = rgbToHex(fill[0], fill[1], fill[2]);
+    styleObj["fill-opacity"] = convertAlphaRange(fill[3]);
+  }
+
+  if (defined(symbol.outline)) {
+    const stroke = symbol.outline;
+    styleObj["stroke"] = rgbToHex(
+      stroke.color[0],
+      stroke.color[1],
+      stroke.color[2]
+    );
+    styleObj["stroke-opacity"] = convertAlphaRange(stroke.color[3]);
+    styleObj["stroke-width"] = stroke.width;
+  }
+
+  if (defined(symbol.imageData)) {
+    styleObj["marker-url"] = `data:${symbol.contentType};base64,${
+      symbol.imageData
+    }`;
+  }
+
+  return styleObj;
+}
+
+function convertAlphaRange(value) {
+  return value / 255;
+}
+
 function loadGeoJson(item) {
   return loadJson(buildGeoJsonUrl(item)).then(function(json) {
     // console.log('Converting first element of', json.layers);
     return featureDataToGeoJson(json.layers[0]);
   });
 }
 
+function loadMetadata(item) {
+  const metaUrl = buildMetdataUrl(item.url);
+  return loadJson(metaUrl).then(function(json) {
```

Comment:
It might be necessary to proxy if the server doesn't support CORS, use `proxyCatalogItemUrl`.